### PR TITLE
Update rubocop gem and configuration.

### PR DESCRIPTION
**Pull request by riccardo-giomi**
## Changes proposed in this pull request:
1. Update Gemfile and configuration for Rubocop
2. Hide current rubocop offenses in .rubocop_todo.yml with `rubocop --auto-gen-config`

## Ready for review:
@weedySeaDragon, @patmbolger

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -114,6 +114,13 @@ group :development, :test do
   gem 'spring-commands-rspec'
 end
 
+group :development, :test do
```
Comment (weedySeaDragon) [inline Gemfile]: Should put these in the _existing_ `group :development, :test` section

File: Gemfile [hunk 2]
```diff
@@ -114,6 +114,13 @@ group :development, :test do
...
+  gem 'rubocop',             require: false
```
Comment (weedySeaDragon) [inline Gemfile]: specify the version for each of these so we're all consistent
Ex: For me, the version of `rubocop` for this project is `0.75.0`,  reports an error with the `.rubocop.yml` file

File: spec/helpers/admin_only/dashboard_helper_spec.rb [hunk 3]
```diff
@@ -220,7 +220,7 @@
           expect(helper.styled_item_then_text('css-class',
                                               'leading item',
                                               'some text',
-                                              spacer: '--')).to match /<span class=\"css-class\">leading item<\/span>(.)+some text/
+                                              spacer: '--')).to match(/<span class=\"css-class\">leading item<\/span>(.)+some text/)
```
Comment (weedySeaDragon) [inline spec/helpers/admin_only/dashboard_helper_spec.rb:223]: revert this change and put it into the `.rubocop_todo.yml` file

**Review by weedySeaDragon — CHANGES_REQUESTED** [review]
minor changes

Looks like the `poltergeist` gem is holding back the version of `rubocop` to 0.78 or less.  (approx.)  Then the `rubycritic` gem has a conflict with 0.89.  
We can do a separate, simple PR (and PT chore) that removes those from the Gemfile so that this can be resolved.
- we're not using the poltergeist gem
- developers can install the rubycritic gem on their own (e.g. globally, not in this project)

File: spec/helpers/admin_only/dashboard_helper_spec.rb (hunk 3 above)
Comment (riccardo-giomi) [reply spec/helpers/admin_only/dashboard_helper_spec.rb:223]: For some reason this is the only offense that is not ignored if I put it into the .rubocop_todo.yml file. That's why I kept the auto-correction.

File: Gemfile (hunk 1 above)
Comment (riccardo-giomi) [reply Gemfile]: Done. I missed that section somehow...

File: Gemfile (hunk 2 above)
Comment (riccardo-giomi) [reply Gemfile]: I fixed all versions in the Gemfile, for now I kept rubocop at 0.75.0:

  * rubocop (0.75.0)
  * rubocop-performance (1.6.1)
  * rubocop-rails (2.5.2)
  * rubocop-rspec (1.35.0)

**Comment by weedySeaDragon** [comment]
Looking good!

**Review by weedySeaDragon — APPROVED** [review]
